examples/dataset_example/src/dataset_example.h

```diff
@@ -0,0 +1,170 @@
+        },
+          "coordinates": ["inline", "crossline", "depth", "cdp-x", "cdp-y"],
+          "compressor": {"name": "blosc", "algorithm": "zstd"}
+        },
+        {
+          "name": "velocity",
+          "dataType": "float16",
+          "dimensions": ["inline", "crossline", "depth"],
+          "metadata": {
+            "chunkGrid": {
+              "name": "regular",
+              "configuration": { "chunkShape": [128, 128, 128] }
+            },
+            "unitsV1": {"speed": "m/s"}
+          },
+          "coordinates": ["inline", "crossline", "depth", "cdp-x", "cdp-y"]
+        },
+        {
+          "name": "image_inline",
+          "dataType": "float32",
+          "dimensions": ["inline", "crossline", "depth"],
+          "longName": "inline optimized version of 3d_stack",
+          "compressor": {"name": "blosc", "algorithm": "zstd"},
+          "metadata": {
+            "chunkGrid": {
+              "name": "regular",
+              "configuration": { "chunkShape": [4, 512, 512] }
+            }
+          },
+          "coordinates": ["inline", "crossline", "depth", "cdp-x", "cdp-y"]
+        },
+        {
+          "name": "image_headers",
+          "dataType": {
+            "fields": [
+              {"name": "cdp-x", "format": "int32"},
+              {"name": "cdp-y", "format": "int32"},
+              {"name": "elevation", "format": "float16"},
+              {"name": "some_scalar", "format": "float16"}
+            ]
+          },
+          "dimensions": ["inline", "crossline"],
+          "metadata": {
+            "chunkGrid": {
+              "name": "regular",
+              "configuration": { "chunkShape": [128, 128] }
+            }
+          },
+          "coordinates": ["inline", "crossline", "cdp-x", "cdp-y"]
+        },
+        {
+          "name": "inline",
+          "dataType": "uint32",
+          "dimensions": [{"name": "inline", "size": 256}]
+        },
+        {
+          "name": "crossline",
+          "dataType": "uint32",
+          "dimensions": [{"name": "crossline", "size": 512}]
+        },
+        {
+          "name": "depth",
+          "dataType": "uint32",
+          "dimensions": [{"name": "depth", "size": 384}],
+          "metadata": {
+            "unitsV1": { "length": "m" }
+          }
+        },
+        {
+          "name": "cdp-x",
+          "dataType": "float32",
+          "dimensions": [
+            {"name": "inline", "size": 256},
+            {"name": "crossline", "size": 512}
+          ],
+          "metadata": {
+            "unitsV1": { "length": "m" }
+          }
+        },
+        {
+          "name": "cdp-y",
+          "dataType": "float32",
+          "dimensions": [
+            {"name": "inline", "size": 256},
+            {"name": "crossline", "size": 512}
+          ],
+          "metadata": {
+            "unitsV1": { "length": "m" }
+          }
+        }
+      ]
+    })";
+  return ::nlohmann::json::parse(schema);
+};
```

Comment:
Nit-pick: Function curly brackets shouldn't end with semicolon.
Lint: Fine should end with new-line.

We don't run linting and formatting CI/CD tests against the examples directory.
```suggestion
}

```